client/me/concierge/info-step.js

```diff
@@ -4,38 +4,129 @@
  * External dependencies
  */
 import React, { Component } from 'react';
+import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
+import moment from 'moment-timezone';
 
 /**
  * Internal dependencies
  */
-import Button from 'components/button';
 import CompactCard from 'components/card/compact';
+import FormButton from 'components/forms/form-button';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormSettingExplanation from 'components/forms/form-setting-explanation';
+import FormTextarea from 'components/forms/form-textarea';
+import Timezone from 'components/timezone';
 import PrimaryHeader from './primary-header';
 import Site from 'blocks/site';
+import { localize } from 'i18n-calypso';
+import { updateConciergeInfo } from 'state/concierge/info/actions';
+import { getConciergeInfo } from 'state/selectors';
 
 class InfoStep extends Component {
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			timezone: moment.tz.guess(),
+			message: '',
+		};
+
+		// use them settings from redux if we have
+		if ( props.info ) {
+			this.state.timezone = props.info.timezone;
+			this.state.message = props.info.message;
+		}
+	}
+
 	static propTypes = {
+		info: PropTypes.object,
 		onComplete: PropTypes.func.isRequired,
 		site: PropTypes.object.isRequired,
 	};
 
+	setFieldValue = attribute => {
+		let name, value;
+		if ( attribute.target ) {
```

Comment:
My personal preference is — since `<Timezone>` doesn't pass the same `event` object as normal input change handlers get, instead of putting a conditional in here to capture that, create two separate functions. Even with the comment below, it's confusing why we're checking for `.target` unless you understand already that `<Timezone>`'s handler doesn't have one.

```
setTimezone = timezone => this.setState( { timezone } );

setFieldValue = ( { event: { target: { name, value } } } ) => this.setState( { [ name ]: value } );
```

Something like that. It reads a lot cleaner to me and is much more obvious right away what each one does.

Reply:
Definitely more clean, updated.